nautobot/dcim/forms.py

```diff
@@ -2253,6 +2253,11 @@ class DeviceFilterForm(
         label="Has a primary IP",
         widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
     )
+    has_interfaces = forms.NullBooleanField(
```

Comment:
I have to add this to make DynamicGroupModelTest pass since DG derives the filters from filterset_form.

Reply:
Something seems off here. You didn't change the `has_interfaces` filter definition, so why did this need to be added? And are there a bunch of other filters that also need to be added/updated in the forms now?

Reply:
`has_interfaces` is a special case, since it was the one that sparked the “we shouldn’t shadow related names and change their intent/type” initiative when it was originally named `interfaces`. All of the other `has_foo` filters were added later if they didn't already exist. For `DeviceFilterSet`, however, I think this is the only one that didn't already have a `has_foo` form field because it would have been confusing in v1.x.